### PR TITLE
Task/cooks 67

### PR DESCRIPTION
**Pull request by taoteg**
## Overview: ##

Collection of many tweaks and improvements to the dashboard components.

## Related Jira tickets: ##

* [COOKS-67](https://jira.tacc.utexas.edu/browse/COOKS-67)

## Summary of Changes: ##

- Refactored the Report and Analysis Chart components.

-  Destructured the ConfigurableChart component into four new plot components:
- - MaltreatmentTypesPlot
- - ObservedFeaturesPlot
- - PredictiveFeaturesPlot
- - FeaturesSummaryPlot (TBD)

- Added better instructions and verbiage to UI instructions

- Factored out DebugPlot component

- Factored out ChartInstructions component

## Testing Steps: ##
1. Build & Run project.

### CODE REVIEW

File: client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+import {
+  getMaltreatmentAggregatedValue,
+  getMaltreatmentSelectedValues
+} from '../util';
+import DebugPlot from './DebugPlot';
+import './MaltreatmentTypesPlot.css';
+
+// Set this to true to inspect the component data in a tabular view.
+// This will hide the chart rendering.
+const debugState = false;
```
Comment (nathanfranklin) [inline client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:14]: you could have this as a property of MaltreatmentTypesPlot and have its default value set to `false`

File: client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      const traceFillColor = plotCategoryColors[i];
+      const type = getBarVertTrace(yData, xData, tName, traceFillColor);
+      newPlotData.push(type);
+    }
+    return newPlotData;
+  };
+
+  // Assign Plot Variables.
+
+  const plotCategoryColors = [
```
Comment (nathanfranklin) [inline client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:117]: future refactor: could define these for the two different plots to do.

File: client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    maltreatmentTypesDataAggregateMaltreatment,
+    maltreatmentTypesListMaltreatment,
+    plotStateMaltreatment
+  ) => {
+    return (
+      <div className="maltreatment-types-plot-layout">
+        <div className="maltreatment-types-plot-header">
+          <div className="maltreatment-types-plot-info">
+            {/* <div className="maltreatment-types-plot-info-item">
+              <div className="maltreatment-types-plot-selected-map">
```
Comment (nathanfranklin) [inline client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:197]: drop?

**Review by nathanfranklin — APPROVED** [review]
LGTM!  Thanks for talking through next steps before merging.

File: client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js (hunk 1 above)
Comment (taoteg) [reply client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:24]: I like this more as I think of it. Going to create a ticket to fix that later.

File: client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js (hunk 2 above)
Comment (taoteg) [reply client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:127]: Yeah, making a ticket to pull a number of methods out of the Plots into a plot helper module.

File: client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js (hunk 3 above)
Comment (taoteg) [reply client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:207]: Keeping for now (in case they want it back in their design).

Comment (taoteg) [reply client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js:207]: SCratch that, dropping it anyway. =) Backed up in a  local branch.

**Pull request merged**